pkgs/by-name/li/libloot/package.nix

```diff
@@ -0,0 +1,166 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+
+  rustPlatform,
+  rust-cbindgen,
+
+  cmake,
+  pkg-config,
+
+  doxygen,
+  python3,
+
+  boost,
+  fmt_11,
+  gtest,
+  icu,
+  spdlog,
+  tbb_2021_11,
+  yaml-cpp,
+}:
+
+let
+  testing-plugins = fetchFromGitHub {
+    owner = "Ortham";
+    repo = "testing-plugins";
+    rev = "refs/tags/1.6.2";
+    hash = "sha256-3Aa98EwqpuGA3YlsRF8luWzXVEFO/rs6JXisXdLyIK4=";
+  };
+
+  buildRustFFIPackage =
+    args:
+    rustPlatform.buildRustPackage (
+      args
+      // {
+        postConfigure = ''
+          cp -r --no-preserve=all ${testing-plugins} testing-plugins
+        '';
+
+        nativeBuildInputs = [ rust-cbindgen ];
+
+        buildAndTestSubdir = "ffi";
+
+        postBuild = ''
+          cbindgen ffi/ -l $lang -o $out/include/$header
+        '';
+      }
+    );
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "libloot";
+  version = "0.24.1";
+
+  src = fetchFromGitHub {
+    owner = "loot";
+    repo = "libloot";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-Y0YYPlfCkXW+ONmXZ/C/jV/VEkagXzt3qHhN5zjxBUw=";
+  };
+
+  patches = [ ./deps.patch ];
+
+  postPatch = ''
+    substituteInPlace src/api/plugin.{h,cpp} \
+        --replace-fail 'Vec_PluginMetadata' 'Vec<::PluginMetadata>'
+  '';
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+
+    doxygen
+    python3.pkgs.sphinx
+    python3.pkgs.sphinx-rtd-theme
+    python3.pkgs.breathe
+  ];
+
+  buildInputs = [
+    boost
+    fmt_11
+    gtest
+    icu
+    (spdlog.override { fmt = fmt_11; })
+    tbb_2021_11
+    yaml-cpp
+
+    finalAttrs.passthru.libloadorder
+    finalAttrs.passthru.esplugin
+    finalAttrs.passthru.loot-condition-interpreter
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeFeature "FETCHCONTENT_SOURCE_DIR_TESTING-PLUGINS" "${testing-plugins}")
+    (lib.cmakeFeature "ESPLUGIN_LIBRARIES" "esplugin_ffi")
+    (lib.cmakeFeature "LIBLOADORDER_LIBRARIES" "loadorder_ffi")
+    (lib.cmakeFeature "LCI_LIBRARIES" "loot_condition_interpreter_ffi")
+  ];
+
+  postBuild = ''
+    sphinx-build -b html ../docs docs/html
+  '';
```

Comment:
Is this expensive? Maybe it should be optional (e.g. a `withDocs` arg)?

Reply:
This takes ~2 seconds.
Though I extracted it out into a separate output, this way the main output takes up less storage.

Reply:
I also ended up adding a `withDocs` option, so that we don't depend on python.
This way we I was able to cross compile the library to aarch64-linux

Reply:
Oh, it turns out using `python3Packages` instead of `python3.pkgs` fixed cross compiling... In that case I'll enable docs by default.